Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -144,7 +147,7 @@ public async Task ValidateConnectionStringAsync(string connectionString)
             if (!match.Groups[CONNSTRING_REGEX_KEY].Value.IsNullOrWhiteSpace())
             {
                 this.ValidateConnectionString(connectionString);
-                await this.TestIoTHubReadPermissionsAsync(connectionString);
+                this.TestIoTHubReadPermissionsAsync(connectionString);
```

Comment:
missing await

Reply:
CreateQuery was a function that couldn't be waited on. That is why I had to remove the await. Since we are using it no longer, added back the await